garrysmod/lua/includes/modules/draw.lua

```diff
@@ -113,6 +113,62 @@ function SimpleTextOutlined(text, font, x, y, colour, xalign, yalign, outlinewid
 
 end
 
+--[[---------------------------------------------------------
+	Name: MultiColorText(font, x, y, xalign, yalign, vararg)
+	Desc: Simple draw text at position, but with multiple colors.
+	Usage: vararg is color tables with r/g/b/a elements followed by a string to be drawn with said color.
+-----------------------------------------------------------]]
+function MultiColorText( font, x, y, xAlign, yAlign, ... )
+	
+	font	= font		or "DermaDefault"
+	x		= x			or 0
+	y		= y			or 0
+	xAlign	= xAlign	or TEXT_ALIGN_LEFT
+	yAlign	= yAlign	or TEXT_ALIGN_TOP
+
+	surface.SetFont( font )
+
+	local ipairs, table, origin = ipairs({ ... })
+	local w, h = 0, 0
+
+	if ( xAlign != TEXT_ALIGN_LEFT ) then
+		local fullText = ""
+		
+		for k, v in ipairs, table, origin do
+			if not IsColor( v ) then
+				fullText = fullText .. tostring( v )
+			end
+		end
+
+		w, h = surface.GetTextSize( fullText )
+
+		if ( xAlign == TEXT_ALIGN_CENTER ) then
+			x = x - w/2
+		elseif ( xAlign == TEXT_ALIGN_RIGHT ) then
+			x = x - w
+		end
+	else
+		h = draw.GetFontHeight( font )

```

Comment:
Would be great if it won't use other draw functions.
surface.GetTextSize works as well and use in FontHeight

Reply:
I could use `select(2, surface.GetTextSize("W"))` which is what the draw function uses, but `draw.GetFontHeight()` caches the results so it's better.

Reply:
Fine, don't use select.
I was suggesting `local _, h = surface.GetTextSize(text)`

Reply:
Performance-wise, Wouldn't `draw.GetFontHeight()` still be better because it caches the values?